Formula/hysteria.rb

```diff
@@ -0,0 +1,32 @@
+class Hysteria < Formula
+  desc "Feature-packed proxy & relay tool optimized for lossy, unstable connections"
+  homepage "https://github.com/HyNetwork/hysteria"
+  url "https://github.com/HyNetwork/hysteria.git",
+    tag:      "v1.2.2",
+    revision: "3d54cb43afab6bb700a35ef77b5436af5587470a"
+  license "MIT"
+  head "https://github.com/HyNetwork/hysteria.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w -X main.appVersion=#{version} -X main.appDate=#{time.rfc3339} -X main.appCommit=#{Utils.git_short_head}"
+    execpath = bin/name
+    system "go", "build", "-trimpath", *std_go_args(output: execpath, ldflags: ldflags), "./cmd"
+  end
+
+  service do
+    run [bin/"hysteria", "--config", "#{etc}/hysteria/config.json"]
+    run_type :immediate
+    keep_alive true
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/hysteria --version | cut -d ' ' -f3")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks, I just fix this.